rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -92,16 +122,23 @@ function _M.init_worker()
 end
 
 function _M.call()
-  ngx_balancer.set_more_tries(1)
+  local phase = ngx.get_phase()
+
+  if phase == "balancer" then
+    ngx_balancer.set_more_tries(1)
 
-  local host, port = balance()
+    local host, port = balance()
 
-  local ok
-  ok, err = ngx_balancer.set_current_peer(host, port)
-  if ok then
-    ngx.log(ngx.INFO, "current peer is set to " .. host .. ":" .. port)
+    local ok, err = ngx_balancer.set_current_peer(host, port)
+    if ok then
+      ngx.log(ngx.INFO, "current peer is set to " .. host .. ":" .. port .. " using lb_alg " .. tostring(get_current_lb_alg()))
+    else
+      ngx.log(ngx.ERR, "error while setting current upstream peer to: " .. tostring(err))
+    end
+  elseif phase == "log" then
+    after_balance()
   else
-    ngx.log(ngx.ERR, "error while setting current upstream peer to: " .. tostring(err))
+    return error('must be called in balancer or log, but was called in: ' .. phase)
```

Comment:
nitpick: please use double quote for string here to be consistent with the the rest of code base.